source/fundamentals/connection/tuning.txt

```diff
@@ -0,0 +1,356 @@
+
+Properly sizing your connection pool is critical to your application's
+performance. Consider the following factors:
+
+- Application Server CPU
+- Database Server CPU
+- Database Server Disk I/O
+
+Application Server CPU
+++++++++++++++++++++++
+
+A CPU can process one operation at a time per core. If your CPU has eight
+cores, it can process eight operations in parallel. More operations will
+result in the underlying operating system scheduling operations and
+"sharing time" between operations.
+
+.. example::
+
+   The driver is processing two queries at the same time, and the underlying
+   operating system is scheduling these operations on the same CPU core.
+
+   If each operations takes ~100ms to process, executing these operations
+   concurrently takes more time than executing them sequentially. This is
+   because the cost of pausing and resuming operations isn't free.
+
+Concurrency vs. Parallelism
+```````````````````````````
+
+The terms "concurrency" and "parallelism" are often used interchangeably,
+but they mean different things.
+
+Concurrency is the coordination and execution of multiple tasks and is
+achievable on a single core by sharing time between tasks.
+
+Parallelism is the execution of multiple tasks **simultaneously** and
+requires multiple cores.
+
+.. example::
+
+   Consider birds in a nest waiting to be fed.
+
+   A parent bird brings back food but can only feed one baby at a time.
+   The parent repeatedly brings food back to the nest, ensuring all babies
+   are fed. This is (very slow!) concurrency.
+
+   If there are two parent birds bringing food back, two babies can be fed
+   at the same time. This is parallelism.
+
+   One last note about concurrent operations. When all we need is the **final result**
+   of two tasks, processing them concurrently usually isn't as efficient as
+   processing them sequentially. In other scenarios, such as such as playing
+   chess against a bot while listening to music, concurrency is highly desireable.
+
+Database Server CPU
++++++++++++++++++++
+
+You should have the same considerations as above, but also keep in mind
+that each database server accepts connections from **every** application
+server.
+
+Database Server Disk I/O
+++++++++++++++++++++++++
+
+The two general disk types in use are hard disk drives (HDD) and
+solid-state drives (SSD).
+
+HDDs have spinning disks, called "platters", and a read/write head called
+a "spindle". Locations on disk are accessed by the platters spinning and
+the spindle moving to the correct location. During read and write operations
+from disk, HDDs can only perform operations sequentially.
+
+SSDs are made up of physical circuits, similar to standard computer memory.
+They offer much faster read and write times because there are no platters
+to spin and no spindle to move. Additionally, they can parallelize reads and
+writes.
+
+If using HDDs, it may be beneficial to increase the ``maxPoolSize`` by
```

Comment:
[suggestion] Front-load "ifs," specify the benefit 

If you're using HDDs with a :wikipedia:`RAID <RAID>` array, it may [improve read/write performance?] to increase...